src/main/java/address/Browser/BrowserManager.java

```diff
@@ -1,53 +1,193 @@
 package address.browser;
 
+import address.model.ModelPerson;
+import address.model.Person;
 import com.teamdev.jxbrowser.chromium.Browser;
 import com.teamdev.jxbrowser.chromium.EditorCommand;
 import com.teamdev.jxbrowser.chromium.dom.By;
 import com.teamdev.jxbrowser.chromium.dom.DOMElement;
 import com.teamdev.jxbrowser.chromium.dom.events.DOMEventType;
 import com.teamdev.jxbrowser.chromium.events.FinishLoadingEvent;
 import com.teamdev.jxbrowser.chromium.events.LoadAdapter;
+import javafx.collections.ObservableList;
+import javafx.scene.control.TabPane;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 /**
  * Manages the browser.
  */
 public class BrowserManager {
 
-    private Browser browser;
+    public static final String INITIAL_PAGE_HTML_CODE =
+            "<html><body><h3>To view contact's web page, click on the contact on the left.</h3></body></html>";
+
+    private static final int NUMBER_OF_PRELOADED_PAGE = 3;
+
+    private TabBrowser browser;
+
+    private ObservableList<ModelPerson> filteredModelPersons;
 
-    public BrowserManager() {
-        this.browser = new Browser();
+    public BrowserManager(ObservableList<ModelPerson> filteredModelPersons) {
+        this.filteredModelPersons = filteredModelPersons;
+        browser = new TabBrowser(NUMBER_OF_PRELOADED_PAGE);
         registerListeners();
     }
 
     private void registerListeners() {
-        this.browser.addLoadListener(new LoadAdapter() {
-            @Override
-            public void onFinishLoadingFrame(FinishLoadingEvent finishLoadingEvent) {
-                automateClickingAndScrolling();
-            }
-        });
+        for (int i = 0; i < browser.getNumberOfPages(); i++) {
+            browser.getPage(i).addLoadListener(new LoadAdapter() {
+                @Override
+                public void onFinishLoadingFrame(FinishLoadingEvent finishLoadingEvent) {
+                    if (finishLoadingEvent.isMainFrame()) {
+                        automateClickingAndScrolling(finishLoadingEvent.getBrowser());
+                        System.out.println("finish loading frame");
+                    }
+                }
+            });
+        }
     }
 
     /**
-     * Frees resources allocated to the browser.
+     * Initialises browser to initial state.
```

Comment:
I think the header comment can be better. Maybe:
```
Initialises browser (no need to initial state as it is implied)

This includes:
- Loads instruction page on browser
```

We can add the list if there is anything else to initialise, in which then we abstract the load instruction page into another method.